compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -483,7 +483,11 @@ class PlainPrinter(_ctx: Context) extends Printer {
           else
             Text()
 
-        nodeName ~ "(" ~ elems ~ tpSuffix ~ ")" ~ node.pos.toString
+        val text = nodeName ~ "(" ~ elems ~ tpSuffix ~ ")"
+        if (ctx.settings.Yprintpos.value)
+          text ~ node.pos.toString
+        else
+          text
       case _ =>
```

Comment:
Just FYI, there's a more fluid way to do this:

    nodeName ~ "(" ~ elems ~ tpSuffix ~ ")" ~ (node.pos.toString provided ctx.settings.Yprintpos.value)